### PR TITLE
Renames in mint module

### PR DESCRIPTION
**Pull request by mxz42**
part of #1342

The first commit does renames agreed upon in #1342 .

The second commit renames `OutputConfirmationSignatures` to `MintOutputSignatureShare`. according to the comment above the struct, it will contain (at some point) only the `tbs::BlindedSignatureShare` and the `tbs::BlindedMessage` will be remembered by the mint. If you don't agree I remove the second commit.

```rust
// FIXME: optimize out blinded msg by making the mint remember it
/// Blind signature share from one Federation peer for a single [`MintOutput`]
#[derive(Debug, Clone, Eq, PartialEq, Hash, Deserialize, Serialize, Encodable, Decodable)]
pub struct MintOutputSignatureShare(
    pub TieredMulti<(tbs::BlindedMessage, tbs::BlindedSignatureShare)>,
);
```

thoughts @dpc @elsirion 

--- 

I rebased this branch on #1389, otherwise some tests failed with `No gateway` or something similar.

### CODE REVIEW

**Comment by dpc** [comment]
Probably want to rebase.

**Comment by mxz42** [comment]
you mean rebase on master or should I just wait until #1389 is merged?

**Comment by dpc** [comment]
image

Seems like much. Do you need all of them?

File: modules/fedimint-mint/src/lib.rs
Comment (elsirion) [inline modules/fedimint-mint/src/lib.rs:362]: :+1:

**Review by elsirion — APPROVED** [review]

**Pull request merged**
